### PR TITLE
added docs for azure blob

### PR DESCRIPTION
**Pull request by IsharaPannila**
Relates to https://github.com/kedacore/keda/issues/154

### CODE REVIEW

File: content/scalers/azure-storage-blob.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
++++
+fragment = "content"
+weight = 100
+title = "Azure Storage Blob"
+background = "light"
++++
+
+Scale applications based on Azure Storage Blobs.
+
```
Comment (tomkerkhove) [inline content/scalers/azure-storage-blob.md]: We also provide additional information such as when it's available and whom maintains it:
```suggestion

**Availability:** v1.1+ | **Maintainer:** Community

```

Can we list you as owner of this scaler? Or do you want to support it as Microsoft @jeffhollan?

File: content/scalers/azure-storage-blob.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+spec:
+  scaleTargetRef:
+    deploymentName: azureblob-function
+  triggers:
+  - type: azure-blob
+    metadata:
+      # Required
+      containerName: functionsblob
+      # Required: connection OR authenticationRef that defines connection
+      connection: STORAGE_CONNECTIONSTRING_ENV_NAME # Default: AzureWebJobsStorage. Reference to a connection string in deployment
```
Comment (tomkerkhove) [inline content/scalers/azure-storage-blob.md]: You're specifying both a connection here, as well as trigger authentication which doesn't really make sense - Would you mind adapting so it only uses the trigger authentication object?

File: content/scalers/azure-storage-blob.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+### Trigger Specification
+
+This specification describes the `azure-blob` trigger for Azure Storage Blob.
+
+```yaml
+triggers:
+  - type: azure-blob
+    metadata:
+      containerName: functions-blob # Required: Container name
```
Comment (tomkerkhove) [inline content/scalers/azure-storage-blob.md]: I presume this is aimed at the storage container, not runtime container?

Might be better to clarify this:
```suggestion
      containerName: functions-blob # Required: Name of Azure Blob Storage container
```

Or even rename it to `blobContainerName`?

File: content/scalers/azure-storage-blob.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      blobCount: '5' # Optional. Blob count target for HPA. Default: 5 blobs
```
Comment (tomkerkhove) [inline content/scalers/azure-storage-blob.md]: What does this mean? Amount of blobs to scale on or?

Maybe this is more clear.
```suggestion
      blobCount: '5' # Optional. Amount of blobs to scale out on. Default: 5 blobs 
```

File: content/scalers/azure-storage-blob.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  - type: azure-blob
+    metadata:
+      containerName: functions-blob # Required: Container name
+      blobCount: '5' # Optional. Blob count target for HPA. Default: 5 blobs 
+      connection: STORAGE_CONNECTIONSTRING_ENV_NAME
+      blobPrefix:  # Optional. Prefix for the Blob. Use this to specifiy sub path for the blobs if required. Default : ""
+      blobDelimiter: # Optional. Delimiter for identifying the blob Prefix. Default: "/"
+```
+
+The `connection` value is the name of the environment variable your deployment uses to get the connection string. This is usually resolved from a `Secret V1` or a `ConfigMap V1` collections. `env` and `envFrom` are both supported.
```
Comment (tomkerkhove) [inline content/scalers/azure-storage-blob.md]: I would leave this out as we now also support `TriggerAuthentication`

File: content/scalers/azure-storage-blob.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
++++
+fragment = "content"
+weight = 100
+title = "Azure Storage Blob"
```
Comment (tomkerkhove) [inline content/scalers/azure-storage-blob.md]: I think they are typically called `Azure Blob Storage` or `Azure Storage - Blobs`
```suggestion
title = "Azure Blob Storage"
```

File: content/scalers/azure-storage-blob.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+background = "light"
++++
+
+Scale applications based on Azure Storage Blobs.
```
Comment (tomkerkhove) [inline content/scalers/azure-storage-blob.md]: I think they are typically called `Azure Blob Storage` or `Azure Storage - Blobs`
```suggestion
Scale applications based on Azure Blob Storage.
```

**Comment by tomkerkhove** [comment]
Thanks for the PR @IsharaPannila! I've added a few comments to improve it from a reading/explanatory perspective.

Would you mind having a look?

File: content/scalers/azure-storage-blob.md (hunk 2 above)
Comment (IsharaPannila) [reply content/scalers/azure-storage-blob.md]: Sure. I'll update the code to use only trigger authentication and update this doc as well.

File: content/scalers/azure-storage-blob.md (hunk 3 above)
Comment (IsharaPannila) [reply content/scalers/azure-storage-blob.md]: yeah. better rename it to "blobContainerName" for the clarity. I'll update the code and then update this doc.

File: content/scalers/azure-storage-blob.md (hunk 5 above)
Comment (IsharaPannila) [reply content/scalers/azure-storage-blob.md]: Agreed. I'll first do the changes to work only with TriggerAuthentication and later remove this connection section.

File: content/scalers/azure-storage-blob.md (hunk 1 above)
Comment (IsharaPannila) [reply content/scalers/azure-storage-blob.md]: I'm fine with any decision (list me as owner or Microsoft). Anyway we are going to use this in couple of heavy blob processing projects soon and I'm always available to give the feedback and help to update/maintain this.

Comment (tomkerkhove) [reply content/scalers/azure-storage-blob.md]: Well in that case I'd be happy to keep you as a maintainer, but don't want to pressure you!

If you want, feel free to use `**Maintainer:** Community (@IsharaPannila)`

File: content/scalers/azure-storage-blob.md (hunk 2 above)
Comment (tomkerkhove) [reply content/scalers/azure-storage-blob.md]: No worries, you don't need to change the code to only support trigger auth - My remark is more that this example uses a mixed approach where I think it's better to use either the environment variable approach on ScaledObject **OR** TriggerAuthentication approach

File: content/scalers/azure-storage-blob.md (hunk 5 above)
Comment (tomkerkhove) [reply content/scalers/azure-storage-blob.md]: Ah now I get it - No, please keep this around as it should support both TriggerAuthentication & this approach.

Sorry for the confusion!

Comment (IsharaPannila) [reply content/scalers/azure-storage-blob.md]: Sure. No worries!

File: content/scalers/azure-storage-blob.md (hunk 3 above)
Comment (IsharaPannila) [reply content/scalers/azure-storage-blob.md]: I've updated the code and rename the param to `blobContainerName`. Also updated the docs. Please review again.

Comment (tomkerkhove) [reply content/scalers/azure-storage-blob.md]: Thanks!

File: content/scalers/azure-storage-blob.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+spec:
+  scaleTargetRef:
+    deploymentName: azureblob-function
+  triggers:
+  - type: azure-blob
+    metadata:
+      # Required
+      blobContainerName: functionsblob
+      # Required: connection OR authenticationRef that defines connection
+      connection: STORAGE_CONNECTIONSTRING_ENV_NAME # Default: AzureWebJobsStorage. Reference to a connection string in deployment
```
Comment (tomkerkhove) [inline content/scalers/azure-storage-blob.md]: I would remove this or at least comment it out since it's confusing now.

Comment (IsharaPannila) [reply content/scalers/azure-storage-blob.md]: done. Please review.

**Pull request merged**